client/src/components/graph/util.ts

```diff
@@ -127,3 +144,58 @@ export function createModelTF() {
   mat3.translate(m, m, [-0.5, -0.5]);
   return m;
 }
+
+export function getSpatialPrefixUrl(s3URI: string) {
+  const url = getSpatialUrl(s3URI);
+
+  return `${url}spatial_files/`;
+}
+
+export function getSpatialTileSources(s3URI: string) {
+  const url = getSpatialUrl(s3URI);
+
+  return `${url}spatial.dzi`;
+}
+
+function getSpatialUrl(s3URI: string) {
+  const datasetVersionId = getDatasetVersionId(s3URI);
+
+  const { hostname } = window.location;
+
+  // TODO(thuang): Take rdev into account
+
+  if (hostname.includes("dev")) {
+    return `https://cellxgene.dev.single-cell.czi.technology/spatial-deep-zoom/${datasetVersionId}/`;
+  }
+
+  if (hostname.includes("staging")) {
+    return `https://cellxgene.staging.single-cell.czi.technology/spatial-deep-zoom/${datasetVersionId}/`;
+  }
+
+  if (hostname.includes("prod")) {
+    return `https://cellxgene.cziscience.com/spatial-deep-zoom/${datasetVersionId}/`;
+  }
+
+  return `http://localhost:5005/d/${datasetVersionId}.cxg/static/deep_zoom/`;
```

Comment:
Currently BE is hard coded to generate `super-cool-spatial` deep zoom assets for any dataset. However, FE will only enable spatial mode IF "spatial" is a valid embedding dropdown option